tools/kms_secrets_encryption.py

```diff
@@ -0,0 +1,548 @@
+        key_rings_list = list(map(lambda key_ring: key_ring.name.rpartition('/')[2], response))
+
+        return key_rings_list
+
+
+    def read_tfvars_data(self, tfvars_file):
+        """A method that reads terraform.tfvars for all configuration data.
+
+        This method reads a terraform.tfvars file for all the user-provided 
+        configuration data above the secrets.
+        Args:
+            tfvars_file (str): a path to a terraform.tfvars file
+        Returns:
+            dict: key value pairs for all the terraform.tfvars data
+        """
+
+        tf_data = {}
+
+        with open(tfvars_file, 'r') as f:
+            for line in f:
+                if "# Secrets below can be encrypted using GCP-KMS, refer to the README for more information." in line:
+                    break
+
+                if line[0] in ('#', '\n'):
+                    continue
+            
+                # Split string using the first delimiter
+                key, value = map(str.strip, line.split('=', 1))
+                tf_data[key] = value.replace("\"", "")
+                
+        return tf_data
+
+
+    def read_tfvars_secrets(self, tfvars_file):
+        """A method that reads terraform.tfvars for all secrets.
+
+        This method reads a terraform.tfvars file for all the secrets into a dictionary.
+
+        Args:
+            tfvars_file (str): a path to a terraform.tfvars file
+        Returns:
+            dict: key value pairs for all the terraform.tfvars secrets
+        """
+
+        tf_secrets = {}
+        begin_reading_secrets = False
+
+        with open(tfvars_file, 'r') as f:
+            for line in f:
+                if "# Secrets below can be encrypted using GCP-KMS, refer to the README for more information." in line:
+                    begin_reading_secrets = True
+                    continue
+                
+                if line[0] in ('#', '\n'):
+                    continue
+
+                if begin_reading_secrets:
+                    # Split string using the first delimiter
+                    key, value = map(str.strip, line.split('=', 1))
+                    tf_secrets[key] = value.replace("\"", "")
+                
+        return tf_secrets
+
+
+    def write_new_tfvars(self):
+        """A method that writes a new terraform.tfvars file
+
+        This method writes a new terraform.tfvars file that is ready to be used by
+        Terraform after encrypting or decrypting. 
+        """
+
+        lines = []
+        
+        with open(self.tfvars_path, 'r') as f:
+            for line in f:
+                # Skip copying kms_cryptokey_id to tfvars if it is decrypting
+                if "kms_cryptokey_id" in line:
+                    continue
+
+                lines.append(line.rstrip())
+
+                # Append kms_cryptokey_id to tfvars if it is encrypting
+                if "# must be in the same project and have KMS decryptor permissions for this key." in line:
```

Comment:
dun think it's a good idea to check for such a specific string. any updates to the comments will break this